src/cpp/src/logit_processor.hpp

```diff
@@ -319,19 +306,19 @@ class LogitProcessor {
                 if (sampling_params.top_k > 0) {
                     m_logit_transformers.emplace_back(new LogitTransformers::TopKFilter(sampling_params.top_k));
                 }
+                // TODO: Temperature transorm already does normalization and top_p/top_k do not seem to require it\
+                // It looks obsolete
                 m_logit_transformers.emplace_back(new LogitTransformers::ProbabilityNormalizeTransform());
```

Comment:
Let's check in HF or LLM to avoid of extra code. I have saved this transformer according old sampler code..

Reply:
I didn't see explicit normalization done in vLLM sampler and in our code this transformer does not look useful. 
If there's no top_p and top_k then it's just repeating the last part of temperature transformer. If we apply top filters and don't think applying normalization second time on limited set will be helpful. 

Honestly I wonder if we need normalization at all in this case. After temperature is applied the only transforms we have are top_p and top_k that do not modify logits and after normalization the relations between logits don't change (if we sort them we get the same outcome). But perhaps I'm missing something.